app/Models/UserModel.php

```diff
@@ -2,10 +2,85 @@
 
 namespace App\Models;
 
+use CodeIgniter\Shield\Authentication\Authenticators\Session;
 use CodeIgniter\Shield\Models\UserModel as ShieldUser;
 use App\Entities\User;
 
 class UserModel extends ShieldUser
 {
     protected $returnType = User::class;
+
+    protected function initialize(): void
+    {
+        parent::initialize();
+
+        // Merge properties with parent
+        $this->allowedFields = array_merge($this->allowedFields, [
+            'handed',
+            'thread_count',
+            'post_count',
+            'avatar',
+            'country',
+            'timezone',
+        ]);
+    }
+
+    public function searchMembers(array $search, int $page, int $perPage, string $sortColumn, string $sortDirection): ?array
+    {
+        $selects = [
+            'users.*',
+            '(thread_count + post_count) AS count',
+        ];
+
+        $results = $this
+            ->withIdentities()
+            ->select(implode(', ', $selects))
+            ->join('auth_groups_users', 'auth_groups_users.user_id = users.id', 'left')
+            ->when(isset($search['username']) && $search['username'] !== '', static function ($query) use ($search) {
```

Comment:
Not saying you need to change it, but newer versions of PHP 8 allow new function types that are scoped to the outer function, like: 

```suggestion
            ->when(isset($search['username']) && $search['username'] !== '', fn ($query) => $query->like('users.username', $search['username'], 'both');
```

Or something like that.

Reply:
Will do. I've recently spent some time with older PHP 7 projects, so the new syntax doesn't come naturally.